hs-ucan/library/Web/DID/Types.hs

```diff
@@ -107,21 +101,16 @@ instance FromHttpApiData DID where
       Right val -> Right val
 
 instance Display DID where -- NOTE `pk` here is base2, not base58
-  textDisplay (DID method pk) = header <> UTF8.toBase58Text (BS.pack multicodecW8)
+  textDisplay (Key pk) = "did:key:z" <> UTF8.toBase58Text (BS.pack multicodecW8)
     where
-      header :: Text
-      header = "did:" <> textDisplay method <> ":" <> "z"
-
       multicodecW8 :: [Word8]
       multicodecW8 =
         case pk of
-          Ed25519PublicKey ed  -> 0xed : 0x01 : BS.unpack (BA.convert ed)
-          RSAPublicKey     rsa -> 0x00 : 0xF5 : 0x02 : BS.unpack (BS64.decodeLenient . encodeUtf8 $ textDisplay rsa)
-                               {-   ^     ^     ^
-                                    |     |     |
-                                    |    "expect 373 Bytes", encoded in the mixed-endian format
-                                  "raw"
-                              -}
```

Comment:
Buuuuut, we've removed the old format generation code. If we put this back in we can merge this into main I think.

Reply:
Awesome; let's put it back!

We actually did this with the Oldstyle DIDs last time, too. It worked well! IIRC we "just" wrapped calls to emitted DIDs in `DID.Oldstyle`, and it serialized in the old format. After the browser was able to handle the new format, we removed that wrapper, but kept the ability to deserialize old UCANs.

Does that make sense?

Reply:
Yup! I've taken a close look at `Oldstyle` recently, actually. It's still there and we're using it to announce the fission server's DID in old-style IIRC.
I think in that discussion we realized we can just remove the ... "old oldstyle" and replace it with something else, since webnative doesn't actually "read" the DID, it only takes it verbatim as the audience for its UCANs. But that's something for another PR - I'll put that into an issue.